code/weapon/emp.h

```diff
@@ -87,7 +87,7 @@ void emp_process_ship(ship *shipp);
 
 // start the emp effect for MYSELF (intensity == arbitrary intensity variable, time == time the effect will last)
 // NOTE : time should be in seconds
-void emp_start_local(float intensity, float time);
+void emp_start_local(float intensity, float time, SCP_string text = "");
```

Comment:
Use `const SCP_string &` to avoid unnecessary string copies.

Reply:
This cannot be const or I can't copy the Emp string into it if length is 0

Reply:
Eh whatever. I don't care that much. I'll just add additional unnecessary code to make you happy.

Reply:
A cascading if() statement would avoid the need for copying the Emp string.